pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/BrokerInterceptors.java

```diff
@@ -102,6 +108,78 @@ public void beforeSendMessage(Subscription subscription,
         }
     }
 
+    @Override
+    public void consumerCreated(ServerCnx cnx,
+                                 Consumer consumer,
+                                 Map<String, String> metadata) {
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            value.consumerCreated(
+                    cnx,
+                    consumer,
+                    metadata);
+        }
+    }
+
+    @Override
+    public void producerCreated(ServerCnx cnx, Producer producer,
+                                 Map<String, String> metadata){
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            value.producerCreated(cnx, producer, metadata);
+        }
+    }
+
+    @Override
+    public void messageProduced(ServerCnx cnx, Producer producer, long startTimeNs, long ledgerId,
+                                 long entryId, Rate rateIn,
+                                 Topic.PublishContext publishContext) {
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            value.messageProduced(cnx, producer, startTimeNs, ledgerId, entryId, rateIn, publishContext);
+        }
+    }
+
+    @Override
+    public  void messageDispatched(ServerCnx cnx, Consumer consumer, long ledgerId,
+                                   long entryId, ByteBuf headersAndPayload) {
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            value.messageDispatched(cnx, consumer, ledgerId, entryId, headersAndPayload);
+        }
+    }
+
+    @Override
+    public void messageAcked(ServerCnx cnx, Consumer consumer,
+                              CommandAck ackCmd) {
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            value.messageAcked(cnx, consumer, ackCmd);
+        }
+    }
+
+    @Override
+    public boolean delegateSuperUserCheck(){
+        for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
+            if (value.delegateSuperUserCheck()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isSuperUser(PulsarService pulsarService, String appId, String originalPrincipal){
```

Comment:
I think this would be more suitable to custom authorization provider, where there is already the `isSuperUser()` method.

Reply:
Removed from the changeset